app/controllers/concerns/foreman/controller/user_self_editing.rb

```diff
@@ -0,0 +1,9 @@
+module Foreman::Controller::UserSelfEditing
+  extend ActiveSupport::Concern
+
+  protected
+
+  def editing_self?
```

Comment:
Seems like this isn't very useful, it forces you to change `editing_self?`, and completely duplicate the method every time you use it (changing the sliced params).

How about:

```
def editing_self?
  @editing_self ||= User.current.editing_self?(editing_self_params)
end

def editing_self_params
  params.slice(:controller, :action, :id)
end
```

Then on SshKeysCommon,

```
def editing_self_params
  params.slice(:controller, :action).merge { :id => params[:id] } 
end
```

At that point I think you can also safely revert the changes made here to the `editing_self?` method and leave it unchanged. Any further implementation just needs to provide its own params instead of overriding the entire action